lib/book_settings.py

```diff
@@ -144,12 +143,7 @@ def search_for_asin(self, query):
         except Exception as e:
             try:
                 self._aConnection.close()
-                if self._proxy:
-                    self._aConnection = HTTPSConnection(self._https_address, self._https_port)
-                    self._aConnection.set_tunnel('www.amazon.com', 443)
-                else:
-                    self._aConnection = HTTPSConnection('www.amazon.com')
-
+                self._aConnection.connect()
```

Comment:
While doing a quick manual test I found out this isn't needed - and if you look at the code it is obvious (see `auto_open` in https://github.com/python-git/python/blob/master/Lib/httplib.py).  Whether you want to explicitly reconnect or not is up to you, not sure it adds much but it doesn't hurt either (the auto-open doesn't seem to swallow any exceptions or anything, so I think it is exactly equivalent).